scripts/wod.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+//pool, explode-at, rote
+//where rote means the intial failed dice are rerolled
+//notably, those rerolled dice cannot explode, and failed exploding dice from qualities can't either
+//meaning that rote has to be counted separately AFTER accounting for dice explosion
+//All 10s explode
+//Okay new idea, rote just adds some more dice that are checked for success at the report, 
+//but not checked by explode or anything. 
+
+const SUCCESS_THRESHOLD = 8;
+const DEFAULT_EXPLODE_TARGET = 10;
+
+module.exports = function(robot) {
+	robot.respond(/roll (\d+)(?:e(\d+))?(r?)/i, function (msg) {
+		let answer;
+		let pool = parseInt(msg.match[1]);
+  	let explode_target = parseInt(msg.match[2]) || DEFAULT_EXPLODE_TARGET;
+  	let is_rote = Boolean(msg.match[3]);
+  	if (pool < 1) {
+  		answer = 'That is too few dice, my human';
+  	} else if (pool > 100){
+  		answer = 'That is too many dice, my human';
+  	} else if (explode_target < 8) {
+  		answer = 'That is an invalid explode target, human';
+  	} else if (is_rote) {
+  		let initial_roll = rote(rollMany(pool));
+  		let explode_allowed = (explode(rollMany(pool), explode_target));
```

Comment:
Agh, you're right. But what you have is still broken. So, for a rote roll, you need to do this:

Normal roll with no explosions. Divide result into fails and successes. Apply explosions to successes, reroll fails.

So perhaps something like this:

```
let initial_pool = rollMany(pool);
let successes, failures = splitPool(initial_pool);
let pool_a = explode(successes);
let pool_b = rote(failures);
let return = report(pool_a.concat(pool_b));
```

For some `splitPool` that returns `[successArray, faliureArray]`.